src/swish/websocket.ss

```diff
@@ -246,21 +302,47 @@
                (set! pp 0)
                (r! bv start n)]
               [else
-               (bytevector-copy! payload pp bv start count)
+               (unless (and (eq? bv payload) (fx= pp start))
+                 (bytevector-copy! payload pp bv start count))
                (set! pp (+ pp count))
                count]))]))
       (make-custom-binary-input-port "websocket payload" r! #f #f #f))
 
     (define (extract type payloads)
+      (define default-buf-size 1024)
       (define (if-eof x value)
         (if (eof-object? x)
             value
             x))
-      (let ([ip (payloads->input-port (reverse payloads))])
+      (let ([ip (payloads->input-port payloads)])
         (if (eq? type 'text)
-            (if-eof (get-string-all (binary->utf8 ip)) "")
+            (match-let* ([(,bv0 . ,_) payloads])
```

Comment:
Is this just `(let ([bv0 (car payloads)]) ...)`?

Reply:
You're right, that is clearer.